src/main/scala/cromwell/binding/values/WdlArray.scala

```diff
@@ -30,4 +30,8 @@ case class WdlArray(wdlType: WdlArrayType, value: Seq[WdlValue]) extends WdlValu
       case _ => Failure(new UnsupportedOperationException("Can only TSV serialize an Array[Primitive]"))
     }
   }
+
+  override def collect[T <: WdlValue](filterFn: PartialFunction[WdlValue, T]): Seq[T] = {
+    value flatMap { _.collect(filterFn) }
+  }
```

Comment:
Agreed, I picked that name because it takes a partial function like collect but it's not really a collect. More like a `findAll` maybe ?